### ISSUE TITLE
Use ccache 4.6+ with MSVC

### ISSUE THREAD

**Issue by justinpryzby** [mesonbuild/meson#10423]
ccache since v4.6 supports compiling with MSVC.
Currently meson requires setting "CC=ccache.exe cl".
ccache should be used automatically (in preference to sccache?) the same as for other compilers.

https://github.com/mesonbuild/meson/blob/1c52ac4e156c2f21d44cf800b112c5ea26878185/mesonbuild/compilers/detect.py#L588

**Comment by eli-schwartz** [comment]
This either depends on #10029 or is a duplicate of it, depending on your point of view.

**Comment by anarazel** [comment]
It's not clear to me that it's a good idea to use ccache automatically with msvc, given the debuginfo format implications. That's not to say that it shouldn't be easier, it definitely should.

**Comment by xclaesse** [comment]
Note that sccache on Windows is makes meson configuration terribly slow. The reason is we disable ccache by setting CCACHE_DISABLE in the env when doing compiler checks but that's not being respected by at least sccache, an spawning extra process for every compiler checks is pathetically slow on that OS. I have a fix as part of https://github.com/mesonbuild/meson/pull/10149, commit https://github.com/mesonbuild/meson/pull/10149/commits/8ccdbb0d8db08fd3a0fd3a88c249b83503437243.

**Comment by xclaesse** [comment]
The reason I'm saying this here, is I guess CCACHE_DISABLE would still spawn ccache even if it's no-op. It's probably enough to make the whole configuration way too slow to be acceptable on Windows. But please test it to confirm or not my suspicion.

**Comment by lazka** [comment]
At least with gcc on Windows, configuring glib with ccache takes 26.646s , without ccache 25.122s

**Comment by anarazel** [comment]
The way meson currently configures PCH files unfortunately makes the combination with ccache not work, at least on windows - the problem is that ccache tries to preprocess the .c file, but that fails because the c_pch file is not on the search path. Not sure if it's worth opening a separate issue for that...
